sparkmagic/sparkmagic/auth/customauth.py

```diff
@@ -0,0 +1,30 @@
+﻿from hdijupyterutils.ipywidgetfactory import IpyWidgetFactory
+from sparkmagic.utils.constants import WIDGET_WIDTH
+
+class Authenticator(object):
+    """Base class for implementing an authentication provider for SparkMagic"""
+    def __init__(self):
+        self.url = 'http://example.com/livy'
+        self.widgets = self.get_widgets(WIDGET_WIDTH)
+
+    def get_widgets(self, widget_width):
```

Comment:
Let's add docstrings for each of these functions that describe what they do, and what behavior subclasses should have.